packages/core/src/fhir-to-cda/cda-templates/commons.ts

```diff
@@ -0,0 +1,192 @@
+export function buildCodeCVFromCodeableConcept(
+  codeableConcept: CodeableConcept | undefined
+): CDACodeCV | Entry {
+  if (!codeableConcept) {
+    return withoutNullFlavorString(codeableConcept);
+  }
+
+  const primaryCoding = codeableConcept.coding?.[0];
+
+  const baseCE = primaryCoding
+    ? buildCodeCE({
+        code: primaryCoding.code,
+        codeSystem: primaryCoding.system,
+        codeSystemName: undefined,
+        displayName: primaryCoding.display,
+      })
+    : {};
+
+  const translations = (codeableConcept.coding?.slice(1) || []).map(coding =>
+    buildCodeCE({
+      code: coding.code,
+      codeSystem: coding.system,
+      codeSystemName: undefined,
+      displayName: coding.display,
+    })
+  );
+
+  const codeCV: CDACodeCV = {
+    ...baseCE,
+    originalText: codeableConcept.text,
+    translation: translations?.length ? translations : undefined,
+  };
+
+  return codeCV;
+}
+
+export function buildInstanceIdentifier({
+  root,
+  extension,
+  assigningAuthorityName,
+}: {
+  root?: string | undefined;
+  extension?: string | undefined;
+  assigningAuthorityName?: string | undefined;
+}): CDAInstanceIdentifier {
+  const identifier: CDAInstanceIdentifier = {};
+  if (root) identifier[rootAttribute] = root;
+  if (extension) identifier[extensionAttribute] = extension;
+  if (assigningAuthorityName) identifier[assigningAuthorityNameAttribute] = assigningAuthorityName;
+
+  return identifier;
+}
+
+export function buildInstanceIdentifiersFromIdentifier(
+  identifiers?: Identifier | Identifier[] | undefined
+): CDAInstanceIdentifier[] | Entry {
+  if (!identifiers) {
+    return withNullFlavor(undefined, rootAttribute);
+  }
+
+  const identifiersArray = Array.isArray(identifiers)
+    ? identifiers
+    : identifiers
+    ? [identifiers]
+    : [];
+  return identifiersArray.map(identifier =>
+    buildInstanceIdentifier({
+      root: identifier.system,
+      extension: identifier.value,
+      assigningAuthorityName: identifier.assigner?.display,
+    })
+  );
+}
+
+export function buildTelecom(telecoms: ContactPoint[] | undefined): CDATelecom[] {
+  if (!telecoms) {
+    return [];
+  }
+  return telecoms.map(telecom => ({
+    ...withoutNullFlavorObject(telecom.use, useAttribute),
+    ...withoutNullFlavorObject(telecom.value, valueAttribute),
+  }));
+}
+
+export function buildAddress(address?: Address[]): CDAAddress[] | undefined {
+  return address?.map(addr => ({
+    ...withoutNullFlavorObject(addr.use, useAttribute),
+    streetAddressLine: withoutNullFlavorString(addr.line?.join(" ")),
+    city: withoutNullFlavorString(addr.city),
+    state: withoutNullFlavorString(addr.state),
+    postalCode: withoutNullFlavorString(addr.postalCode),
+    country: withoutNullFlavorString(addr.country),
+    useablePeriod: {
+      low: withoutNullFlavorObject(addr.period?.start, valueAttribute),
+      high: withoutNullFlavorObject(addr.period?.end, valueAttribute),
+    },
+  })); // Using only first address
```

Comment:
What does this mean? Looking at this code, we're mapping all addresses to CDAAddress, and the result is an array.